front-end/src/app/same-different/same-different.component.ts

```diff
@@ -12,8 +12,47 @@ export class SameDifferentComponent implements OnInit {
 
   constructor() { }
 
-  ngOnInit() {}
+  ngOnInit() {
+    this.currentFace = this.facePaths[this.progress];
+    this.randomFace = this.currentFace;
+    if (Math.random() > .5) {
+      this.randomFace = this.facePaths[Math.floor(Math.random() * this.facePaths.length)];
+    }
+  }
 
+  progress : number = 0;
   score : number = 0;
+  stage : number = 0; // memorize, choose, feedback
```

Comment:
A Stage enum would be better than numbers so it's easy for anyone to pick up the code and understand it whenever the code references the stage variable. Like `this.stage = Stage.MEMORIZE` is easier to understand than `this.stage = 0`.

Reply:
Done